test cases/common/99 subproject subdir/meson.build

```diff
@@ -30,6 +30,11 @@ assert(not d.found(), 'Dependency should be not-found')
 d = dependency('sub_implicit')
 assert(d.found(), 'Should implicitly fallback')
 
+# `native: true` should not make a difference when doing a native build.
+if not meson.is_cross_build()
+  dependency('sub_implicit', native: true)
+endif
```

Comment:
I sneaked this here because I was too lazy to open a new test case. But now that you created one better move this there. You don't need a subproject to test override_dependency().

Reply:
okay, moved. Hopefully the dependency I picked works, lol